src/modules/kvs/kvstxn.c

```diff
@@ -495,6 +495,7 @@ static int kvstxn_append (kvstxn_t *kt, json_t *dirent,
         /* entry not found, treat like normal insertion */
         if (treeobj_insert_entry (dir, final_name, dirent) < 0)
             return -1;
+        (*append) = true;
```

Comment:
Not knowing this code well (or at all), a comment here as to why `*append = true` would be helpful, especially given the comment just above (`treat like normal insertion`).